src/sandbox.ml

```diff
@@ -282,13 +282,28 @@ let detect_opam_sandbox ~project_root opam () =
   Opam (opam, switch)
 ;;
 
+let detect_dune_pkg ~project_root () =
+  let open Promise.Syntax in
+  (* Path to the dune.lock file *)
+  let dune_lock_path = Path.join project_root (Path.of_string "dune.lock") in
+  let+ exists = Fs.exists (Path.to_string dune_lock_path) in
+  if exists
+  then (
+    show_message
+      `Info
+      "Dune Pkg detected. Configuring a custom sandbox for the workspace.";
+    Some (Custom "$prog $args"))
```

Comment:
Custom was a handy workaround so far, but we really need to introduce a new variant `Dune_pkg` now.